pkg/controller/operands/kubevirt.go

```diff
@@ -294,35 +295,36 @@ func (h *kvConfigHooks) addEnabledFeatureGates(req *common.HcoRequest, foundFgSp
 
 func filterOutDisabledFeatureGates(req *common.HcoRequest, foundFgSplit []string, resultFg []string) (bool, []string) {
 	fgChanged := false
+	featureGateChecks := getFeatureGateChecks(req.Instance.Spec.FeatureGates)
 	for _, fg := range foundFgSplit {
-		// Remove if not in HC CR
-		switch fg {
-		case HotplugVolumesGate:
-			if !req.Instance.Spec.FeatureGates.IsHotplugVolumesEnabled() {
-				fgChanged = true
-				continue
-			}
-		case kvWithHostPassthroughCPU:
-			if !req.Instance.Spec.FeatureGates.IsWithHostPassthroughCPUEnabled() {
-				fgChanged = true
-				continue
-			}
-		case kvWithHostModelCPU:
-			if !req.Instance.Spec.FeatureGates.IsWithHostModelCPUEnabled() {
-				fgChanged = true
-				continue
-			}
-		case SRIOVLiveMigrationGate:
-			if !req.Instance.Spec.FeatureGates.IsSRIOVLiveMigrationEnabled() {
-				fgChanged = true
-				continue
-			}
+		if isFeatureGateMissingFrom(featureGateChecks, fg) {
+			fgChanged = true
+			continue
 		}
 		resultFg = append(resultFg, fg)
 	}
 	return fgChanged, resultFg
 }
 
+func isFeatureGateMissingFrom(checks featureGateChecks, featureGate string) bool {
+	if check, isKnown := checks[featureGate]; isKnown {
+		return !check()
+	}
+	return false
+}
+
+type featureGateChecks map[string]func() bool
+
+func getFeatureGateChecks(featureGates *hcov1beta1.HyperConvergedFeatureGates) featureGateChecks {
```

Comment:
I don't have strong opinion about this solution. I'm not sure it's not "too much". 

Regardless, maybe we can add a check for `featureGates == nil` before creating the map.

Reply:
It will cause more allocations than the switch. Not sure if it's a concern though.

Reply:
Yes, and it called twice each time.